R/mwrite.R

```diff
@@ -7,9 +7,45 @@ tocode <- function(l) {
   paste0(names(l), " = ", as.character(l))
 }
 
+# block name is $OMEGA or $SIGMA
+mwrite_matrix <- function(x, block_name) {
+  code <- character(0)
+  for(i in seq_along(x$data)) {
+    datai <-  x$data[[i]]
+    labelsi <- x$labels[[i]]
+    namei <- x$names[[i]]
+    code <- c(code, block_name)
+    header <- "@block"
+    if(namei != "...") {
+      header <- c(header, paste0("@name ", namei))  
+    }
+    if(any(labelsi != "...")) {
```

Comment:
Shouldn't this be "." instead of "..."?

``` r                                                                                                                  
library(mrgsolve)                                                                                                      
#>                                                                                                                     
#> Attaching package: 'mrgsolve'                                                                                       
#> The following object is masked from 'package:stats':                                                                
#>                                                                                                                     
#>     filter                                                                                                          
code <- '                                                                                                              
$OMEGA 1 2 3                                                                                                           
$OMEGA 3 4 5                                                                                                           
'                                                                                                                      
mod <- mcode("foo", code)                                                                                              
#> Building foo ...                                                                                                    
#> done.                                                                                                               
a <- mwrite_yaml(mod, file = NULL)                                                                                     
a$omega$labels                                                                                                         
#> $matrix1                                                                                                            
#> [1] "." "." "."                                                                                                     
#>                                                                                                                     
#> $matrix2                                                                                                            
#> [1] "." "." "."                                                                                                     
```